hpack/table.py

```diff
@@ -0,0 +1,207 @@
+from collections import deque
+import logging
+
+log = logging.getLogger(__name__)
+
+def table_entry_size(name, value):
+    """
+    Calculates the size of a single entry
+
+    This size is mostly irrelevant to us and defined
+    specifically to accomidate memory management for
+    lower level implementions. The 32 extra bytes are
+    considered the "maximum" overhead that would be
+    required to represent each entry in the table.
+
+    See RFC7541 Section 4.1
+    """
+    return 32 + len(name) + len(value)
+
+class HeaderTable(object):
+    """
+    Implements the combined static and dynamic header table
+
+    The name and value arguments for all the functions
+    should ONLY be byte strings (b'') however this is not
+    strictly enforced in the interface.
+
+    See RFC7541 Section 2.3
+    """
+    #: Default maximum size of the dynamic table. See
+    #  RFC7540 Section 6.5.2.
+    DEFAULT_SIZE = 4096
```

Comment:
To be clear, I mean an extra newline immediately following the `DEFAULT_SIZE = 4096` line.